### PR TITLE
Feature: address snyk vulnerabilities

### PR DESCRIPTION
**Pull request by marcelosalloum**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Fix Snyk vulnerabilities by upgrading versions and doing some tweaks to the code.

### Why

To make the project safer.

### CODE REVIEW

File: platform/src/main/java/org/stellar/anchor/platform/payment/observer/circle/CirclePaymentObserverService.java [hunk 1]
```diff
@@ -91,11 +92,16 @@ public void handleCircleNotification(CircleNotification circleNotification)
   public void handleSubscriptionConfirmationNotification(CircleNotification circleNotification)
       throws BadRequestException {
     String subscribeUrl = circleNotification.getSubscribeUrl();
-    if (subscribeUrl == null) {
+    if (StringHelper.isEmpty(subscribeUrl)) {
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/payment/observer/circle/CirclePaymentObserverService.java:95]: 💡 Can we add static import?

```suggestion
    if (isEmpty(subscribeUrl)) {
```

File: platform/src/main/java/org/stellar/anchor/platform/payment/observer/circle/CirclePaymentObserverService.java [hunk 2]
```diff
@@ -91,11 +92,16 @@ public void handleCircleNotification(CircleNotification circleNotification)
...
       throw new BadRequestException(
           "Notification body of type SubscriptionConfirmation is missing subscription URL.");
     }
 
+    // sanitize
+    if (!subscribeUrl.startsWith("https://")) {
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/payment/observer/circle/CirclePaymentObserverService.java:101]: Can we also catch this error in the config validator?

Comment (lijamie98) [reply platform/src/main/java/org/stellar/anchor/platform/payment/observer/circle/CirclePaymentObserverService.java:101]: To allow "HTTPS://", 

```suggestion
    if (!subscribeUrl.toLowercase().startsWith("https://")) {
```

File: service-runner/build.gradle.kts [hunk 3]
```diff
@@ -10,6 +10,7 @@ plugins {
 }
 
 dependencies {
+  implementation(libs.snakeyaml)
```
Comment (lijamie98) [inline service-runner/build.gradle.kts:13]: ❓ Do we use `snakeyaml`?

Comment (marcelosalloum) [reply service-runner/build.gradle.kts:13]: It is a dependency from springboot but the version used by springboot has a vulnerability. I'm manually adding it to make sure we can control the version of this dependency and force springboot to use our version.

File: platform/src/main/java/org/stellar/anchor/platform/payment/observer/circle/CirclePaymentObserverService.java (hunk 2 above)
Comment (marcelosalloum) [reply platform/src/main/java/org/stellar/anchor/platform/payment/observer/circle/CirclePaymentObserverService.java]: Good catch! Addressed in https://github.com/stellar/java-stellar-anchor-sdk/pull/577/commits/37a3c9d7ff16eb14a08fe2cb98b1e89789bd9daa.

File: platform/src/main/java/org/stellar/anchor/platform/payment/observer/circle/CirclePaymentObserverService.java (hunk 1 above)
Comment (marcelosalloum) [reply platform/src/main/java/org/stellar/anchor/platform/payment/observer/circle/CirclePaymentObserverService.java]: I've adopted your suggestion but I'm not a big fan of static imports unless the methods are widely used.

I also think `StringHelper` provides useful context to the function call, while a plain `isEmpty` can be mistaken by a local method.

File: service-runner/build.gradle.kts (hunk 3 above)
Comment (lijamie98) [reply service-runner/build.gradle.kts:13]: Thanks for the explanation. 

Does it make sense to upgrade the springboot dependency version?

Comment (lijamie98) [reply service-runner/build.gradle.kts:13]: Can you also leave a comments in the snake yaml dependency? 

If you think it is too much to upgrade spring boot, it is also very ok with me. 

Please feel free to merge when you think it is ok.

**Review by lijamie98 — APPROVED** [review]

File: service-runner/build.gradle.kts (hunk 3 above)
Comment (marcelosalloum) [reply service-runner/build.gradle.kts:13]: There was no available version of springboot where they update that vulnerability themselves, that's why I had to force the update.

I'll check if they've released a new update and if not, I'll add comments as you suggested 👍

**Pull request merged**